vocabulary-draft/continuous-delivery.md

```diff
@@ -0,0 +1,114 @@
+
+Optional attributes: 
+- **PipelineRun URL**: URL to locate where  pipeline instances are running
+- **PipelineRun Errors**: error field to indicate possible compilation, test, build and package errors.
+
+## Product Composition Events
+
+When the products delivered to users can be, or needs to be, broken down into components (hardware, software, configuration files etc.), the creation of versions of said components and the compositoning of components into products or product subsystems may warrant the use of separate events detailed in this section.
+
+Product Composition Events MUST include the following attributes:
+
+- **Event Type**: the type is restricted to include `CD.ProductComposition` prefix. For example `CD.ProductComposition.ComponentVersionCreated` or `CD.ProductComposition.CompositionCreated`
+
+### Component Version Created
+
+Announces that a new version of a Product Component has been created.
+
+Examples of Component Versions include:
+
+- New version of a software module or application.
+- New version of a document or piece of configuration.
+- New revision of a hardware component.
+
+The purpose of this event would be for systems to start creating Product Compositions / Baseline Versions that include this Component Version.
+
+Component Version Created Events MUST include the following attributes:
+
+- **Component Version ID**: Unique identifier of component version.
+- **Component ID**: Identifier of component of which this is a version.
+
+Optional attributes:
+
+- **Version Name**: Machine-friendly and human-friendly version tag/number/etc. for component version, e.g. “1.4.2-alpha2”.
+- **Component Name**: Machine-friendly and human-friendly name of component, e.g. “c37-backbone-router”.
+- **Component Display Name**: UX/human-friendly display name of component, e.g. “C37 Backbone Router”.
+- **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
+- **Component Track**: States what track (e.g. platform variant (x86_64, ARM etc.), project (e.g. X-DataCenter, X-Cloud etc.) or other) the component, of which this is a version, belongs to.
+
+### Product Composition Created
+
+Announces that a new composition of source-files, artifacts, component versions and other compositions/baseline versions, has been created.
+
+__Note:__ This event description suffers from a missing piece of terminology to represent the “recipe” by that controls what is included in new compositions. E.g. “The latest stable version (LSV) of this application, the LSV of this library, and the LSV of this configuration file”. For now, this will be referred to as a “Recipe” until a better name is given. (ESCI calls this a Baseline, Eiffel likely has a term for this as well).
+
+Product Composition Created Events MUST include the following attributes:
+
+- **Composition ID**: Unique identifier of composition.
+- **Recipe ID**: Identifier of recipe of which this is a composition.
```

Comment:
@erkist ok.. a new concept here.. and here I do the problem of trying to adopt and implement such advanced concept. Imagine that if we want to emit an event that includes a Recipe ID we need to first define what a Recipe is, for that, I think that we need to open a discussion in the **interoperability SIG** to actually define what is that.

Reply:
@salaboy Good point! Do you know how I would go about doing that?

Reply:
@erkist maybe we can ask @e-backmark-ericsson about that.. as I haven't been involved too much with that SIG